codec.go

```diff
@@ -98,13 +99,17 @@ func (cdc *Codec) RegisterInterface(ptr interface{}, opts *InterfaceOptions) {
 	var info = cdc.newTypeInfoFromInterfaceType(rt, opts)
 
 	// Finally, check conflicts and register.
-	cdc.mtx.Lock()
-	{
+	func() {
```

Comment:
Same here about this closure being unnecessary, we could have just added `defer cdc.mtx.Unlock()`